BusyTiming/ListModel.swift

```diff
@@ -6,35 +6,55 @@
 //
 
 import Foundation
+import RealmSwift
 
 class ActivitiesModel  {
 
-    var activity: [Activity] = []
+    var activityBase: Results <Activity>!
     var selectedIndex = 0
     var activityName = ""
     var activityTime  = 0
     
-    func addActivity(nameOfActivity: String, isCompleted:Bool = false, time:Int = 0) {
-        let activityStr = Activity(nameInit: nameOfActivity, selectedInit: isCompleted,
-                                timeInit: time)
-        activity.append(activityStr)
-        if activity.count == 1 {
-            activity[0].isSelected = true
-            activityName = activity[0].name
+    private struct ErrorBase {
+        static let ErrorInitBase = "Realm initialization error"
+        static let ErrorChangeBase = "Base change error"
+    }
+    
+    init() {
+        do{
+            let realm = try Realm()
+            self.activityBase = realm.objects(Activity.self)
+        }
+        catch {
+            print(ErrorBase.ErrorInitBase)
         }
     }
-
-    func changeState (index: Int) {
-        if index != selectedIndex {
-            activity[index].isSelected = !(activity[index].isSelected)
-            activity[selectedIndex].isSelected = false
+    
+    func addActivity(nameOfActivity: String, isCompleted:Bool = false, time:Int = 0) {
```

Comment:
I believe you will not need 'isCompleted'. we don't plan to have incomplete activities.